### PR TITLE
added ruby exercises

### PR DESCRIPTION
**Pull request by jasonglves20**
Completed ruby exercises along with unit tests for each exercise.

### CODE REVIEW

File: ruby/exercise1/test.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'test/unit'
+require_relative 'exercise1'
+
+class TestSerialAverage < Test::Unit::TestCase
+    def test_serial_average
```
Comment (minaxi98) [inline ruby/exercise1/test.rb:5]: test cases for invalid inputs

File: ruby/exercise2/exercise2.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+def element_at(array, index)
```
Comment (minaxi98) [inline ruby/exercise2/exercise2.rb:1]: What if array index is out of bound

File: ruby/exercise2/test.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    assert_equal([1, 2], non_inclusive_range([1, 2, 3], 0, 2))
+    assert_equal(["a", "b"], non_inclusive_range(["a", "b", "c"], 0, 2))
+    assert_equal([1], non_inclusive_range([1, 2, 3], 0, 1))
+  end
+
+  def test_start_and_length
+    assert_equal([1, 2], start_and_length([1, 2, 3], 0, 2))
+    assert_equal(["a", "b"], start_and_length(["a", "b", "c"], 0, 2))
+    assert_equal([2, 3], start_and_length([1, 2, 3], 1, 2))
+  end
```
Comment (minaxi98) [inline ruby/exercise2/test.rb:41]: test cases for edge cases

File: ruby/exercise3/exercise3.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+def check_hash(hash)
+    hash[543121] = "100"
+    hash.each_key do |key|
+        if key.is_a?(Integer)
+            if key.even?
```
Comment (minaxi98) [inline ruby/exercise3/exercise3.rb]: Can be done in one if statement

File: ruby/exercise4/exercise4.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+def skip_sports(array, integer)
+    string_array = []
+    array.each_with_index do |sport, index|
```
Comment (minaxi98) [inline ruby/exercise4/exercise4.rb:6]: Can this be done with for loop?

File: ruby/exercise4/test.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    expected_output = ["3:hockey", "4:soccer"]
+    assert_equal(expected_output, skip_sports(array, 3))
+
+    array = ["tennis", "golf", "swimming", "cycling", "running"]
+    expected_output = ["4:running"]
+    assert_equal(expected_output, skip_sports(array, 4))
+
+    array = ["volleyball", "badminton", "table tennis", "squash", "rugby"]
+    expected_output = []
+    assert_equal(expected_output, skip_sports(array, 5))
```
Comment (minaxi98) [inline ruby/exercise4/test.rb:16]: test cases for index is out of range

File: ruby/exercise6/test.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def teardown
+    # File.delete("jason.txt")
+  end
+
+  def test_base64_encode
+    expected_output = "Originally, John Doe was a sham name used to indicate any plaintiff in an action of ejectment (a legal action to regain property) in civil court [ “ dGhlIHN0cmluZyB0aGF0IHdpbGwgYmUgYmFzZSBlbmNvZGVk ”]. Richard Roe was the counterpart, to indicate the defendant. These fake names were used in delicate legal matters, a practice that was abolished in English law in 1852."
+    actual_output = File.read("jason.txt")
+    assert_equal(expected_output, actual_output)
```
Comment (minaxi98) [inline ruby/exercise6/test.rb]: Handle file exceptions

Comment (minaxi98) [inline ruby/exercise6/test.rb]: Hndle file exceptions

File: ruby/exercise5/exercise5.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+first_name = data["name"].split(" ")[0]
+last_name = data["name"].split(" ")[1]
+
+data.delete("name")
+
+data["first_name"] = first_name
+data["last_name"] = last_name
+
+your_name= "jason"
+
```
Comment (minaxi98) [inline ruby/exercise5/exercise5.rb:18]: Begin rescue can be used for file operation

File: ruby/exercise6/exercise6.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+require 'base64'
+
+txt_file = File.read("content.txt")
```
Comment (minaxi98) [inline ruby/exercise6/exercise6.rb]: This could have been done in class

File: ruby/exercise6/test.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
+require 'test/unit'
+require 'base64'
+require_relative 'exercise6'
+
+class TestBase64Encode < Test::Unit::TestCase
+  def setup
+    @txt_file = File.read("content.txt")
+    @your_name = "jason.txt"
```
Comment (minaxi98) [inline ruby/exercise6/test.rb]: Better to use class and use object here

**Review by minaxi98 — CHANGES_REQUESTED** [review]
